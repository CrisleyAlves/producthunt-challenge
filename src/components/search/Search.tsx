import React from "react";
import clsx from "clsx";

export default function Search({
  showSearchForm,
  onClickClearSearchForm,
  handleOnSubmitSearchForm,
}: {
  showSearchForm: boolean;
  onClickClearSearchForm: () => void;
  handleOnSubmitSearchForm: (e: React.FormEvent<HTMLFormElement>) => void;
}) {
  return (
    <div
      className={clsx("search__wrapper", {
        "search__wrapper--active": showSearchForm,
      })}
    >
      <form onSubmit={handleOnSubmitSearchForm}>
        <input name="topic" className="search__input" />
        <button
          type="reset"
          className="search__clear"
          onClick={onClickClearSearchForm}
        >
          clear
        </button>
      </form>
    </div>
  );
}
