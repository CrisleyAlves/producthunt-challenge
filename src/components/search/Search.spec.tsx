import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";

import Search from "./Search";

describe("Search", () => {
  it("renders the search input and clear button", () => {
    const { container } = render(
      <Search
        showSearchForm={true}
        onClickClearSearchForm={vi.fn()}
        handleOnSubmitSearchForm={vi.fn()}
      />
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByRole("textbox", { name: "" })).toBeDefined();
    expect(screen.getByRole("button", { name: /clear/i })).toBeDefined();
  });

  it("calls onClickClearSearchForm when clear button is clicked", () => {
    const onClickClearSearchForm = vi.fn();
    render(
      <Search
        showSearchForm={true}
        onClickClearSearchForm={onClickClearSearchForm}
        handleOnSubmitSearchForm={vi.fn()}
      />
    );
    fireEvent.click(screen.getByRole("button", { name: /clear/i }));
    expect(onClickClearSearchForm).toHaveBeenCalled();
  });

  it("applies active class when showSearchForm is true", () => {
    const { container } = render(
      <Search
        showSearchForm={true}
        onClickClearSearchForm={vi.fn()}
        handleOnSubmitSearchForm={vi.fn()}
      />
    );
    expect(container).toMatchSnapshot();
    expect(container.firstChild.className).toContain("search__wrapper--active");
  });

  it("does not apply active class when showSearchForm is false", () => {
    const { container } = render(
      <Search
        showSearchForm={false}
        onClickClearSearchForm={vi.fn()}
        handleOnSubmitSearchForm={vi.fn()}
      />
    );
    expect(container.firstChild.className).not.toContain(
      "search__wrapper--active"
    );
  });
});
