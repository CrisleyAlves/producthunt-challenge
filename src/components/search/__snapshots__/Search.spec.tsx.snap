// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Search > applies active class when showSearchForm is true 1`] = `
<div>
  <div
    class="search__wrapper search__wrapper--active"
  >
    <form>
      <input
        class="search__input"
        name="topic"
      />
      <button
        class="search__clear"
        type="reset"
      >
        clear
      </button>
    </form>
  </div>
</div>
`;

exports[`Search > renders the search input and clear button 1`] = `
<div>
  <div
    class="search__wrapper search__wrapper--active"
  >
    <form>
      <input
        class="search__input"
        name="topic"
      />
      <button
        class="search__clear"
        type="reset"
      >
        clear
      </button>
    </form>
  </div>
</div>
`;
